### PR TITLE
Add register new case

### PR DESCRIPTION
**Pull request by digitalronin**
This change implements the registerNewCase GLiMR API call, and includes some refactoring of common code.

### CODE REVIEW

File: lib/glimr_api_client/register_new_case.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      raise RequestError, errors unless errors.blank?
+    end
+
+    def endpoint
+      '/registernewcase'
+    end
+
+    def request_body
+      {
+        contactPhone: @params[:contactPhone],
```
Comment (csutter) [inline lib/glimr_api_client/register_new_case.rb]: Not sure how I feel about this – seems like a big list of duplicated `xyz: @params[:xyz]`.

Would it be worth extracting the key names into an array and then just `#map`ping that?

File: lib/glimr_api_client/api.rb [hunk 2]
```diff
@@ -10,15 +10,20 @@ module Api
         'https://glimr-api.taxtribunals.dsd.io/Live_API/api/tdsapi'
     )
 
+    # TODO: extract error handling method
```
Comment (tatyree) [inline lib/glimr_api_client/api.rb]: Agreed.  

Worth noting that there are two different categories of error handling going on here 😕. The `handle_response_errors` method deals with api server errors while the rescue deals with network errors.

File: lib/glimr_api_client/api.rb [hunk 3]
```diff
@@ -10,15 +10,20 @@ module Api
...
+
     def post
       client("#{DEFAULT_ENDPOINT}#{endpoint}").post(body: request_body.to_query).tap { |resp|
         # Only timeouts and network issues raise errors.
         handle_response_errors(resp)
         @body = resp.body
       }
     rescue Excon::Error => e
-      if endpoint.eql?('/paymenttaken')
+      case endpoint
```
Comment (tatyree) [inline lib/glimr_api_client/api.rb]: 👍

File: lib/glimr_api_client/base.rb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+module GlimrApiClient
```
Comment (tatyree) [inline lib/glimr_api_client/base.rb]: :+1:

File: lib/glimr_api_client/register_new_case.rb [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        errors << required if request_body.fetch(required).blank?
+      end
+      raise RequestError, errors unless errors.blank?
+    end
+
+    def endpoint
+      '/registernewcase'
+    end
+
+    def request_body
```
Comment (tatyree) [inline lib/glimr_api_client/register_new_case.rb]: unk...

Sorry, not helpful...Mainly just meant "wow, that's a lot of body params..."

File: spec/glimr_api_client/register_new_case_spec.rb [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+	let(:api) { described_class.new(params) }
+
+	before do
+		allow(api).to receive(:client).and_return(excon)
+	end
+
+  context 'validating parameters' do
+		let(:valid_params) { { jurisdictionId: 8, onlineMappingCode: 'something' } }
+
+		it 'barfs when no parameters are supplied' do
```
Comment (tatyree) [inline spec/glimr_api_client/register_new_case_spec.rb]: Ok to use something other than 'barfs'?  it 'fails validation when no parameters...' maybe?  For me, 'barfs' is just a bit too colloquial.

File: lib/glimr_api_client/api.rb (hunk 2 above)
Comment (digitalronin) [inline lib/glimr_api_client/api.rb]: Sorry, that "TODO" was for me - I forgot to do it! Will do it now.

File: lib/glimr_api_client/register_new_case.rb (hunk 1 above)
Comment (digitalronin) [inline lib/glimr_api_client/register_new_case.rb]: Yes, very possibly. We could just return @params, here. Any thoughts on that?

File: spec/glimr_api_client/register_new_case_spec.rb (hunk 6 above)
Comment (digitalronin) [inline spec/glimr_api_client/register_new_case_spec.rb]: Sure. I'll change it.

File: lib/glimr_api_client/register_new_case.rb (hunk 1 above)
Comment (csutter) [inline lib/glimr_api_client/register_new_case.rb]: Ah yes – all for that :+1:

File: spec/glimr_api_client/register_new_case_spec.rb [hunk 7]
```diff
@@ -0,0 +1,93 @@
+require 'rails_helper'
```
Comment (csutter) [inline spec/glimr_api_client/register_new_case_spec.rb]: What's up with the indentation here? 😕

**Pull request merged**